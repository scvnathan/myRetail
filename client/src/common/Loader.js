import React from "react";
import logo from "~images/logo.png";
import smallLogo from "~images/logo-sm.png";
import styled, { keyframes } from "styled-components";

const loaderAnim = keyframes`
	0%,47%, 100% { transform: scale3d(1,1,1); }
	10% { transform: scale3d(1.25,.75,1); }
	14% { transform: scale3d(.85,1.15,1); }
	20% { transform: scale3d(1.15,.85,1); }
	24% { transform: scale3d(.95,1.05,1); }
	36% { transform: scale3d(1.05,.95,1); }
`;

const LoaderLogo = styled.img`
	animation: ${loaderAnim};
	animation-duration: 2s;
	animation-timing-function: ease;
	animation-delay: 0s;
	animation-iteration-count: infinite;
	animation-direction: normal;
	animation-fill-mode: both;
`;

const Wrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	margin: 0 auto;
	height: 100vh;
	justify-content: center;
	width: 45%;
`;

export default () => {
	return <LoaderLogo src={smallLogo} />;
};

export function FullScreenLoader() {
	return (
		<Wrapper>
			<LoaderLogo src={logo} />
		</Wrapper>
	);
}
