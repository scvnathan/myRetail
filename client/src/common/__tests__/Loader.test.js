import React from "react";
import Loader, {FullScreenLoader} from "../Loader";
import renderer from "react-test-renderer";

describe("Loader component", () => {
	it("should render correctly", () => {
		const comp = <Loader />;
		const tree = renderer.create(comp).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should render correctly (Fullscreen)", () => {
		const comp = <FullScreenLoader />;
		const tree = renderer.create(comp).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
