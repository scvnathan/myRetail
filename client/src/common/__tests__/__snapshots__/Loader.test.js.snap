// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render correctly (Fullscreen) 1`] = `
<div
  className="Loader__Wrapper-sc-14f76tl-1 BCCQG"
>
  <img
    className="Loader__LoaderLogo-sc-14f76tl-0 hdyJfX"
    src="test-file-stub"
  />
</div>
`;

exports[`Loader component should render correctly 1`] = `
<img
  className="Loader__LoaderLogo-sc-14f76tl-0 hdyJfX"
  src="test-file-stub"
/>
`;
